### PR TITLE
Backlog 20617

### PR DESCRIPTION
**Pull request by kurtwalker**

### CODE REVIEW

**Comment by kurtwalker** [comment]
requires https://github.com/pentaho/pentaho-kettle/pull/4741 to be merged first

**Comment by kurtwalker** [comment]
@pentaho/ackbar please review

File: kettle-plugins/kafka/src/main/java/org/pentaho/big/data/kettle/plugins/kafka/KafkaConsumerInput.java [hunk 1]
```diff
@@ -97,255 +57,36 @@ public KafkaConsumerInput( StepMeta stepMeta, StepDataInterface stepDataInterfac
...
-        .count() > 0;
-    if ( errorsPresent ) {
-      return false;
+
+    kafkaConsumerInputData.outputRowMeta = new RowMeta();
+    try {
+      kafkaConsumerInputMeta.getFields( kafkaConsumerInputData.outputRowMeta,
+        getStepname(), null, null, this, repository, metaStore );
+    } catch ( KettleStepException e ) {
+      // TODO: just log this exception and return false?
```
Comment (DFieldFL) [inline kettle-plugins/kafka/src/main/java/org/pentaho/big/data/kettle/plugins/kafka/KafkaConsumerInput.java]: Do we want to log this exception

File: kettle-plugins/kafka/src/main/java/org/pentaho/big/data/kettle/plugins/kafka/KafkaStreamSource.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+  @Override public void close() {
+    super.close();
+    callable.shutdown();
+    future.cancel( true );
+  }
+
+  @Override public void open() {
+    if ( future != null ) {
+      // TODO: create message property
```
Comment (DFieldFL) [inline kettle-plugins/kafka/src/main/java/org/pentaho/big/data/kettle/plugins/kafka/KafkaStreamSource.java]: Wasn't sure if we want to create a specific message property

File: kettle-plugins/kafka/src/main/java/org/pentaho/big/data/kettle/plugins/kafka/KafkaConsumerInput.java (hunk 1 above)
Comment (kurtwalker) [reply kettle-plugins/kafka/src/main/java/org/pentaho/big/data/kettle/plugins/kafka/KafkaConsumerInput.java]: yup, I'll make that change

File: kettle-plugins/kafka/src/main/java/org/pentaho/big/data/kettle/plugins/kafka/KafkaStreamSource.java (hunk 2 above)
Comment (kurtwalker) [reply kettle-plugins/kafka/src/main/java/org/pentaho/big/data/kettle/plugins/kafka/KafkaStreamSource.java]: I don't think this one needs to be localized

Comment (DFieldFL) [reply kettle-plugins/kafka/src/main/java/org/pentaho/big/data/kettle/plugins/kafka/KafkaStreamSource.java]: 👍

**Pull request merged**